packages/ra-ui-materialui/src/input/ArrayInput/SimpleFormIterator.tsx

```diff
@@ -166,19 +190,39 @@ export const SimpleFormIterator = (props: SimpleFormIteratorProps) => {
                         </SimpleFormIteratorItem>
                     ))}
                 </ul>
-                {!disabled && !disableAdd && (
-                    <div className={SimpleFormIteratorClasses.add}>
-                        {cloneElement(addButton, {
-                            className: clsx(
-                                'button-add',
-                                `button-add-${source}`
-                            ),
-                            onClick: handleAddButtonClick(
-                                addButton.props.onClick
-                            ),
-                        })}
-                    </div>
-                )}
+                <div className={SimpleFormIteratorClasses.line}>
+                    {!disabled && !disableAdd && (
+                        <div className={SimpleFormIteratorClasses.add}>
+                            {cloneElement(addButton, {
+                                className: clsx(
+                                    'button-add',
+                                    `button-add-${source}`
+                                ),
+                                onClick: handleAddButtonClick(
+                                    addButton.props.onClick
+                                ),
+                            })}
+                        </div>
+                    )}
+                    {fields.length > 0 && (
+                        <div className={SimpleFormIteratorClasses.clear}>
+                            {cloneElement(clearButton, {
+                                className: clsx(
+                                    'button-clear',
+                                    `button-clear-${source}`
+                                ),
+                                onClick: () => setIsOpen(true),
+                            })}
+                        </div>
+                    )}
+                </div>
+                <Confirm
```

Comment:
Why put it far from the button?

Reply:
Confirm move close to the button

commit c97dc5569f2b46b401f3446410f98be19b759e48